File: src/main/java/com/shelterhub/dto/request/EstimatedAgeRequest.java
package com.shelterhub.dto.request;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;

@Data
@Builder
public class EstimatedAgeRequest {
    private int years, months, days;

    public LocalDate toEstimatedAge() {
        return LocalDate.now()
                .minusYears(years)
                .minusMonths(months)
                .minusDays(days);
    }
}

File: src/main/kotlin/br/com/shelterhubmanagementapi/dto/request/EstimatedAgeRequest.kt
package br.com.shelterhubmanagementapi.dto.request

import java.time.LocalDate

data class EstimatedAgeRequest(
    val years: Int,
    val months: Int,
    val days: Int
)

fun EstimatedAgeRequest.toEstimatedAge(): LocalDate = LocalDate.of(this.years, this.months, this.days)
